analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -64,8 +64,11 @@ public CSharpSymbolUsageCollector(Compilation compilation, IEnumerable<ISymbol>
 
         public override void Visit(SyntaxNode node)
         {
-            semanticModel = node.EnsureCorrectSemanticModel(semanticModel ?? compilation.GetSemanticModel(node.SyntaxTree));
-            base.Visit(node);
+            semanticModel = node.EnsureCorrectSemanticModelOrDefault(semanticModel ?? compilation.GetSemanticModel(node.SyntaxTree));
+            if (semanticModel != null)
```

Comment:
For the C# usage symbol collector, you can create two different "snippet" compilations in the UT, and give compilation1 to the constructor, and give a SyntaxNode from the second compilation to Visit. Then, it should not do the visiting.

See ConstantValueFinderTest for example how to do that

And then check `UsedSymbols` or some other property, that it collects symbols when the semanticModel is resolved, and that it doesn't when the SemanticModel doesn't get resolved.